### ISSUE TITLE
[Dropdown] a11y announcements

### ISSUE THREAD

**Issue by brentertz** [mineral-ui/mineral-ui#402]
As a mineral-ui consumer, I want the Dropdown component to have better a11y announcement support, so that I can more efficiently understand and navigate the menu using AT.

**AC:**

* 

**Other Considerations:**

* From a11y Slack:
    * Something to keep in mind for react apps is that you may want to use shouldComponentUpdate to prevent the live region from being re-rendered if the text hasn’t changed. Otherwise screen readers will keep reading it repeatedly for every render cycle

**Comment by brentertz** [comment]
Recent updates to Dropdown have vastly improved a11y and keyboard navigation.  The necessity of this story needs validated ideally by someone that typically uses a screen-reader, preferably something in addition to VoiceOver.
